### PR TITLE
Improve File.cp_r/3 regarding callback argument

### PR DESCRIPTION
**Pull request by eksperimental**

### CODE REVIEW

File: lib/elixir/lib/file.ex [hunk 1]
```diff
@@ -670,8 +670,9 @@ defmodule File do
       end
 
   """
-  @spec cp_r(Path.t, Path.t, (Path.t, Path.t -> boolean)) :: {:ok, [binary]} | {:error, posix, binary}
-  def cp_r(source, destination, callback \\ fn(_, _) -> true end) when is_function(callback) do
+  @spec cp_r(Path.t, Path.t, (Path.t, Path.t -> boolean)) ::
+        {:ok, [binary]} | {:error, posix, binary}
+  def cp_r(source, destination, callback \\ fn(_, _) -> true end) when is_function(callback, 2) do
```
Comment (ericmj) [inline lib/elixir/lib/file.ex]: Maybe we should leave the unrelated newline change alone and focus on the docs in this PR. If there are lines that need to be broken up we can do it separately. WDYT?

Comment (ericmj) [inline lib/elixir/lib/file.ex]: I see that the guard is also being changed but I think the newline should stay as-is.

File: lib/elixir/lib/file.ex [hunk 2]
```diff
@@ -634,10 +634,10 @@ defmodule File do
...
   the contents inside source into the destination.
 
-  If a file already exists in the destination,
-  it invokes a callback which should return
-  `true` if the existing file should be overwritten,
-  `false` otherwise. The callback defaults to return `true`.
+  If a file already exists in the destination, it invokes `callback`.
+  `callback` must be a function that takes two arguments: `source` and `destination`,
+  which should return `true` if the existing file should be overwritten, and `false` otherwise.
+  The callback defaults to return `true`.
```
Comment (whatyouhide) [inline lib/elixir/lib/file.ex]: The punctuation looks weird here in the new sentence. What about

> `callback must be a function that takes two arguments: `source` and `destination`. The callback should return `true` if the existing file should be overwritten and `false` otherwise.

?

File: lib/elixir/lib/file.ex [hunk 3]
```diff
@@ -670,8 +670,9 @@ defmodule File do
       end
 
   """
-  @spec cp_r(Path.t, Path.t, (Path.t, Path.t -> boolean)) :: {:ok, [binary]} | {:error, posix, binary}
-  def cp_r(source, destination, callback \\ fn(_, _) -> true end) when is_function(callback) do
+  @spec cp_r(Path.t, Path.t, (Path.t, Path.t -> boolean)) ::
```
Comment (whatyouhide) [inline lib/elixir/lib/file.ex]: Also as @ericmj said, this spec can stay on one line.

File: lib/elixir/lib/file.ex (hunk 2 above)
Comment (eksperimental) [inline lib/elixir/lib/file.ex]: that's perfect. Thank you!

**Comment by eksperimental** [comment]
All comments addressed.

**Comment by whatyouhide** [comment]
Thanks a lot @eksperimental! 💟
